pkg/controller/elasticsearch/user/reconcile.go

```diff
@@ -106,9 +106,12 @@ func aggregateFileRealm(
 	}
 
 	// reconcile predefined users
-	elasticUser, err := reconcileElasticUser(ctx, c, es, existingFileRealm, userProvidedFileRealm, passwordHasher)
-	if err != nil {
-		return filerealm.Realm{}, esclient.BasicAuth{}, err
+	var elasticUser users
+	if !es.Spec.Auth.DisableElasticUser {
+		elasticUser, err = reconcileElasticUser(ctx, c, es, existingFileRealm, userProvidedFileRealm, passwordHasher)
```

Comment:
If you move the conditional check into the `reconcileElasticUser` function then non of the code here needs to change. `reconcileElasticUser` would return and empty  `users` slice and that would return an empty realm.